R/router.R

```diff
@@ -160,11 +173,79 @@ create_router_callback <- function(root, routes) {
   }
 }
 
+#' Create router UI
+#'
+#' Creates router UI in Shiny applications.
+#'
+#' @param default Main route to which all invalid routes should redirect.
+#' @param ... All other routes defined with shiny.router::route function.
+#' @param page_404 Styling of page when wrong bookmark is open. See \link{page404}.
+#' @param env Environment (only for advanced usage).
+#'
+#' @return Application UI wrapped in a router.
+#'
+#' @examples
+#' \dontrun{
+#'   ui <- function() {
+#'     router_ui(
+#'       route("/", root_page(id = "root")),
+#'       route("/other", other_page(id = "other")),
+#'       page_404 = page404(
+#'         message404 = "Please check if you passed correct bookmark name!")
+#'     )
+#'   }
+#' }
+#' @export
+router_ui <- function(default, ..., page_404 = page404(), env = parent.frame()) {
+  routes <- c(default, ...)
+  root <- names(default)[1]
+  if (! PAGE_404_ROUTE %in% names(routes))
+    routes <- c(routes, route(PAGE_404_ROUTE, page_404))
+  router <- list(root = root, routes = routes)
+
+  routes_input_id <- "routes"
+  if (!is.null(env$ns)) {
```

Comment:
I am not sure - this is needed to ensure that `shiny.router` will work when defined inside a module right?

If yes, doesn't this assume that the user defined `ns` in their module?

Wouldn't that break if a user instead of:
```r
innerUI <- function(id) {
  ns <- NS(id)
  "This is the inner UI"
}
```

does something like this:
```r
innerUI <- function(id) {
  my_ns <- NS(id)
  "This is the inner UI"
}
```

I mean assume `ns` is a standard naming convention, but maybe we should document this assumption? e.g. in `@details`:

_If you are defining the router inside a shiny module, we assume that the namespacing function defined in the UI is named as `ns`._

Reply:
Thank you for pointing that out - I am so used to `ns` that I didn't think that someone might use something else!
Updating the documentation seems fine.